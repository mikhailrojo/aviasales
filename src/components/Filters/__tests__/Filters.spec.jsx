import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import {Filters} from '../index';

describe('Snapshot', () => {
	it('render Filters with all options', () => {
		const renderer = new ShallowRenderer();
		const component = renderer.render(
			<Filters
				filterByStops={jest.fn()}
				chosenStops={[false,false,false,true]}
				changeCurrency={jest.fn()}
				selectedCurrency="RUB"
				currencyRates={{
					RUB: 1,
					EUR: 72,
					USD: 63
				}}
			/>
		);
		expect(component).toMatchSnapshot();
	});
	it('render Filters with no currencyRates', () => {
		const renderer = new ShallowRenderer();
		const component = renderer.render(
			<Filters
				filterByStops={jest.fn()}
				chosenStops={[false,false,false,true]}
				changeCurrency={jest.fn()}
				selectedCurrency="RUB"
			/>
		);
		expect(component).toMatchSnapshot();
	});
	it('render Filters with no chosenStops', () => {
		const renderer = new ShallowRenderer();
		const component = renderer.render(
			<Filters
				filterByStops={jest.fn()}
				changeCurrency={jest.fn()}
				selectedCurrency="RUB"
			/>
		);
		expect(component).toMatchSnapshot();
	});
});