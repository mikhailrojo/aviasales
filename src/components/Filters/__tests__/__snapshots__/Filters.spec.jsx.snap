// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot render Filters with all options 1`] = `
<styled.div>
  <CurrencyFilter
    changeCurrency={[Function]}
    selectedCurrency="RUB"
  />
  <StopFilter
    chosenStops={
      Array [
        false,
        false,
        false,
        true,
      ]
    }
    filterByStops={[Function]}
  />
</styled.div>
`;

exports[`Snapshot render Filters with no chosenStops 1`] = `
<styled.div>
  
</styled.div>
`;

exports[`Snapshot render Filters with no currencyRates 1`] = `
<styled.div>
  <StopFilter
    chosenStops={
      Array [
        false,
        false,
        false,
        true,
      ]
    }
    filterByStops={[Function]}
  />
</styled.div>
`;
